csharp/src/Client/AdbcCommand.cs

```diff
@@ -167,8 +167,13 @@ public new AdbcDataReader ExecuteReader(CommandBehavior behavior)
         {
             switch (behavior)
             {
+                case CommandBehavior.SingleResult:
+                case CommandBehavior.KeyInfo:
+                case CommandBehavior.SequentialAccess:
+                case CommandBehavior.CloseConnection:
                 case CommandBehavior.SingleRow:
-                    throw new InvalidOperationException("Cannot read rows in Arrow");
+                    throw new InvalidOperationException($"{behavior} is not supported with this provider");
```

Comment:
This seems backwards to me. Shouldn't this only succeed when CommandBehavior is a supported value instead of failing on explicitly-unsupported values? Also, note that CommandBehavior is a "Flags" enum so combinations of these values are possible too.